src/btree/bt_handle.c

```diff
@@ -531,7 +531,11 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
      * connection-wide base write generation. In that case, we should initialize our write gen just
      * ahead of our btree specific write generation.
      */
-    btree->write_gen = btree->base_write_gen = WT_MAX(ckpt->write_gen + 1, conn->base_write_gen);
+    btree->write_gen = WT_MAX(ckpt->write_gen + 1, conn->base_write_gen);
+    if (!F_ISSET(session, WT_SESSION_IMPORT) && ckpt->run_write_gen > conn->base_write_gen)
```

Comment:
I think this does the trick.

The only part I'm concerned about is if we import and don't do another checkpoint. Will we read the old `run_write_gen` be read (and used since `WT_SESSION_IMPORT` isn't set)?

I think no because the changes to metadata won't be checkpointed so we'd need to import again.

Reply:
That sounds right. Import will rely on a metadata checkpoint completing before it's final.